vertica_python/vertica/connection.py

```diff
@@ -697,19 +699,19 @@ def read_expected_message(self, expected_types, error_handler=None):
 
     def read_bytes(self, n):
         if n == 1:
-            result = self._socket().recv(1)
+            result = self.socket_as_file.read(1)
```

Comment:
Even though this works so far, if presume `self.socket_as_file` is not None (i.e. socket is connected), which is risky if we change the logic in the future. One possible solution would be 
```python
def _socket_as_file(self):
    if not self.socket_as_file:
        self.socket_as_file = self._socket().makefile('rb')
    return self.socket_as_file

def read_bytes(self, n):
    ...
    result = self._socket_as_file().read(1)
```